### PR TITLE
Feat milestone funds allocation

### PR DESCRIPTION
**Pull request by gabyteera**

### CODE REVIEW

**Review by GabijaBeres — APPROVED** [review]

File: src/components/tooltipList/styled.tsx [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
+  }
 
-    & > li:not(:last-child) {
-      margin-bottom: 5px;
-    }
-  
   .completed {
     &::marker {
-      content: '✓   ';
+      content: "✓   ";
```
[the remark this answers is not in the document]
Comment (gabyteera) [reply src/components/tooltipList/styled.tsx:15]: Good point :) milestonesArray is not yet used (we don't have any designs for it at the moment), but later whitespaces could definitely be changed to margins

File: src/web3/calculateFundsAllocation.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  for (let i in milestonesInvestmentsListForFormula) {
+    if (milestonesInvestmentsListForFormula[i] !== 0) {
+      let stream =
+        milestonesInvestmentsListForFormula[i] *
+        (milestones[i].intervalStreamingPortion / percentageDivider);
+      let seed =
+        milestonesInvestmentsListForFormula[i] *
+        (milestones[i].intervalSeedPortion / percentageDivider);
+      let total = stream + seed;
```
Comment (BenasVolkovas) [inline src/web3/calculateFundsAllocation.ts:30]: With a stream, seed and total number, you should use BigNumber functions like `add()` or `mul()`.
Example:
`milestonesInvestmentsListForFormula[i].mul(milestones[i].intervalStreamingPortion).div(percentageDivider);`

File: src/web3/calculateFundsAllocation.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+      let milestoneFundsAllocation = {
+        streamAllocated: ethers.utils.formatEther(
+          BigNumber.from(parseInt(stream.toString()))
+        ),
+        seedAllocated: ethers.utils.formatEther(
+          BigNumber.from(parseInt(seed.toString()))
+        ),
+        totalFundsAllocated: ethers.utils.formatEther(
```
Comment (BenasVolkovas) [inline src/web3/calculateFundsAllocation.ts:39]: streamAllocated, seedAllocated, totalFundsAllocated can format ether straight from big number.
Example
`streamAllocated: ethers.utils.formatEther(stream)`

File: src/web3/calculateFundsAllocation.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      };
+
+      milestonesFundsAllocation.push(milestoneFundsAllocation);
+    } else {
+      let stream =
+        firstNonZero *
+        (milestones[i].intervalStreamingPortion / percentageDivider);
+      let seed =
+        firstNonZero * (milestones[i].intervalSeedPortion / percentageDivider);
+      let total = stream + seed;
```
Comment (BenasVolkovas) [inline src/web3/calculateFundsAllocation.ts:51]: With a stream, seed and total number, you should use BigNumber functions like add() or mul().
Example:
`firstNonZero.mul(milestones[i].intervalStreamingPortion).div(percentageDivider)`

File: src/web3/calculateFundsAllocation.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+      let milestoneFundsAllocation = {
+        streamAllocated: ethers.utils.formatEther(
+          BigNumber.from(parseInt(stream.toString()))
+        ),
+        seedAllocated: ethers.utils.formatEther(
+          BigNumber.from(parseInt(seed.toString()))
+        ),
+        totalFundsAllocated: ethers.utils.formatEther(
```
Comment (BenasVolkovas) [inline src/web3/calculateFundsAllocation.ts:60]: streamAllocated, seedAllocated, totalFundsAllocated can format ether straight from big number.
Example
`streamAllocated: ethers.utils.formatEther(stream)`

File: src/web3/calculateFundsAllocation.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+) => {
+  let milestonesFundsAllocation: IMilestoneFundsAllocated[] = [];
+
+  let firstDupeIndex = milestonesInvestmentsListForFormula.findIndex(
+    (item: number, index: number) =>
+      milestonesInvestmentsListForFormula.lastIndexOf(item) !== index &&
+      item === 0
+  );
+
+  let firstNonZero = milestonesInvestmentsListForFormula[firstDupeIndex - 1];
```
Comment (BenasVolkovas) [inline src/web3/calculateFundsAllocation.ts:20]: Simpler approach
`firstNonZero = milestonesInvestmentsListForFormula.slice().reverse().find((element) => element != 0)`

File: src/hooks/useLoadValues.ts [hunk 7]
```diff
@@ -108,25 +113,37 @@ export const useLoadValues = () => {
...
 
         const milestoneCount = (await contract.getMilestonesCount()).toNumber();
         const currentMilestone = (
           await contract.getCurrentMilestoneId()
         ).toNumber();
 
+        const percentageDivider = Number(await contract.getPercentageDivider());
+        const milestonesInvestmentsList =
+          await contract.getMilestonesInvestmentsListForFormula();
+        for (let i in milestonesInvestmentsList) {
```
Comment (BenasVolkovas) [inline src/hooks/useLoadValues.ts:139]: Maybe we can avoid for loop and assign the list straight away?
`setMilestonesInvestmentListForFormula(milestonesInvestmentsList)`

**Review by BenasVolkovas — CHANGES_REQUESTED** [review]

**Review by BenasVolkovas — APPROVED** [review]

**Pull request merged**